seshdash/utils/reporting.py

```diff
@@ -81,3 +82,31 @@ def prepare_report(site, duration="week"):
 
 def report(subject,content,recipients):
     return send_mail(subject, recipients, content, email_template="reporting")
+
+
+def generate_report(report):
+    """
+    Function to generate a report,
+    The function receives a report model instance and
+    it returns a dict containing the aggregated value of the 
+    report attributes
+    """ 
+    report_data = {}
+    now = datetime.now()
+    
+    # Getting the correct date range
+    if report.duration == "daily":
+        start = now - relativedelta(hours=24)
```

Comment:
Relative delta works the same as timedelta except it has many parameters like hours and months.